packages/dx-react-grid/src/plugins/table-header-row.tsx

```diff
@@ -81,16 +84,28 @@ class TableHeaderRowBase extends React.PureComponent<TableHeaderRowProps> {
                 const groupingEnabled = isColumnGroupingEnabled
                   && isColumnGroupingEnabled(columnName)
                   && atLeastOneDataColumn;
+                const nextColumnName = getNextColumnName(tableColumns, columnName);
+                const lastColumn = getLastColumnName(tableColumns) === columnName;
 
                 return (
                   <HeaderCell
                     {...params}
                     column={params.tableColumn.column!}
                     draggingEnabled={draggingEnabled && atLeastOneDataColumn}
-                    resizingEnabled={tableColumnResizingEnabled}
-                    onWidthChange={({ shift }) => changeTableColumnWidth({ columnName, shift })}
-                    onWidthDraft={({ shift }) => draftTableColumnWidth({ columnName, shift })}
+                    resizingEnabled={
+                      tableColumnResizingEnabled
+                      && (!lastColumn || !nextColumnResizing)
+                    }
+                    onWidthChange={({ shift }) => changeTableColumnWidth({
+                      columnName, nextColumnName, shift,
+                    })}
+                    onWidthDraft={({ shift }) => draftTableColumnWidth({
+                      columnName, nextColumnName, shift,
```

Comment:
Can we use `React.useCallback` here?

Reply:
We can use `React.useCallback` only in functional components. TableHeaderRow is a class.

Reply:
Okay, what about our own `memoize` function?

Reply:
We need this?
We call that function only when we resize column. It is already called optimall (every call we recieve new `shift` property)

Reply:
You are creating a new function on each render. I think this is not a good

Reply:
I'm afraid we can't do it in Table Header Row plugin, cause we have few columns with that function.